### ISSUE TITLE
image preview uses 100% CPU usage and does not preview image.

### ISSUE THREAD

**Issue by jedi2610** [ranger/ranger#2089]
#### Runtime Environment

- Operating system and version: Ubuntu 20.04
- Terminal emulator and version: alacritty 0.6.0-dev
- Python version: 3.8.2 (default, Jul 16 2020, 14:00:26) [GCC 9.3.0]
- Ranger version/commit:  ranger 1.9.3
- Locale: en_IN.ISO8859-1

#### Current Behavior
Image preview does not display any output and keeps loading forever. CPU usage is at 100%. The image is a 6k image. This specifically happens on high res images but not on all high res images.

Image previewer: Ueberzug
previewing the image using Ueberzug alone works fine though.

#### Steps to reproduce
Should i like share the image or something to reproduce the issue?

**Comment by toonn** [comment]
@seebye, does the ueberzug previewing code do more than simply pass a path to ueberzug?

**Comment by seebye** [comment]
No, it just converts a dict to json and writes it to the stdin of ueberzug.

**Comment by toonn** [comment]
Hmm, not sure how to troubleshoot this further. Since ranger should hardly do anything in this scenario and ueberzug works.

**Comment by jedi2610** [comment]
[issue](https://github.com/seebye/ueberzug/issues/113) that i opened in ueberzug

**Comment by toonn** [comment]
@jedi2610, hmm, could you try testing with a bash script that uses ueberzug's silent flag?

**Comment by jedi2610** [comment]
it works normally

**Comment by toonn** [comment]
Next step would involve replacing the call to ueberzug in ranger to just print the arguments that are passed.

**Comment by seebye** [comment]
I think it would be good to mention which process does cause the cpu load.  
`htop -d 200` should display your processes and gives you 200 seconds to copy the names & the arguments of the most cpu intensive commands.

**Comment by jedi2610** [comment]
i actually destroyed my ubuntu installation by mistake while using the `dd` command and ended up reinstalling ubunutu. now it works fine tho. i don't know what caused the problem.

**Comment by toonn** [comment]
Then I'm gonna close this as unresolved since we'd need more info to try to find a solution.

**Comment by jedi2610** [comment]
@toonn it's happening again. but this time on a gif (8.54MB). this time it takes up 100% RAM too.

**Comment by toonn** [comment]
Really weird. Nothing suspicious in top or anything? Basically nothing we can do about this without more info.

**Comment by jedi2610** [comment]
okay what kind of info do you need?

**Comment by toonn** [comment]
All of it : )
There's too many possibilities and I don't know how to narrow it down. First things first, disable previews completely and see whether slowdowns still occur. Next step *if* previews are the cause is to pare down `scope.sh` until we find what's causing it.

**Comment by jedi2610** [comment]
it works normally after turning off the previews.
and here is a link to my scope.sh
https://termbin.com/t4qe

**Comment by toonn** [comment]
Remove all the orientation stuff in the `image/*` clause in `handle_image` so it's just `exit 7;;`.

**Comment by jedi2610** [comment]
i was caught up in work will check it out tomorrow @toonn

**Comment by jedi2610** [comment]
@toonn i did that but that didn't change anything. and now (for like 4 days now) i'm not getting any preview at all for any images or video files. so strange. once again i'm sorry i was caught up in work.

**Comment by toonn** [comment]
You didn't turn off image previews altogether did you? Or configure the path to your `scope.sh` and then throw it away?

**Comment by jedi2610** [comment]
no @toonn but i'll check again tho and let you know

**Comment by jedi2610** [comment]
no @toonn  it's all like exactly the same

**Comment by JuPlutonic** [comment]
@toonn  / @jedi2610 I fixed it on _MacOSX10.15 and Ubuntu18.04 machines_ **X** _1.9.4 and HEAD from master_ by using command:
ranger --cmd="eval inport PIL'

**Comment by toonn** [comment]
Hmm, that's a weird fix, usually means the problem is hard to figure out.
Would appreciate reports from others whether opening ranger that way fixes the problem!
